packages/dcos-integration-test/extra/test_mesos.py

```diff
@@ -171,6 +171,100 @@ def get_ps_output():
+        agent_id = None
+        for framework in state['frameworks']:
+            for task in framework['tasks']:
+                if app_id in task['id']:
+                    agent_id = task['slave_id']
+        assert agent_id is not None, 'Agent ID not found for instance of app_id {}'.format(app_id)
+
+        # Find hostname from agent_id
+        agent_hostname = None
+        for agent in state['slaves']:
+            if agent['id'] == agent_id:
+                agent_hostname = agent['hostname']
+        assert agent_hostname is not None, 'Agent hostname not found for agent_id {}'.format(agent_id)
+        logging.debug('Located %s on agent %s', app_id, agent_hostname)
+
+        # Fetch the Mesos agent statistics
+        r = dcos_api_session.get('/monitor/statistics', host=agent_hostname, port=5051)
+        assert r.status_code == 200
+        stats = r.json()
+
+        total_io_serviced = None
+        total_io_service_bytes = None
+        for stat in stats:
+            # Find the statistic for the Marathon application that we deployed. Since what that
+            # Marathon application launched is a Mesos command task (i.e., using Mesos built-in
+            # command executor), the executor ID will be same as the task ID, so if we find the
+            # `app_id` in an executor ID of a statistic, that must be the statistic entry
+            # corresponding to the application that we deployed.
+            if app_id in stat['executor_id']:
+                # We only care about the blkio throttle statistics but not the blkio cfq statistics,
+                # because in the environment where the disk IO scheduler is not `cfq`, all the cfq
+                # statistics may be 0.
+                throttle_stats = stat['statistics']['blkio_statistics']['throttling']
+                for throttle_stat in throttle_stats:
+                    if 'device' not in throttle_stat:
+                        total_io_serviced = throttle_stat['io_serviced'][0]['value']
+                        total_io_service_bytes = throttle_stat['io_service_bytes'][0]['value']
+
+        assert total_io_serviced is not None, ('Total blkio throttling IO serviced not found '
+                                               'for app_id {}'.format(app_id))
+        assert total_io_service_bytes is not None, ('Total blkio throttling IO service bytes '
+                                                    'not found for app_id {}'.format(app_id))
+        # We expect the statistics retrieved from Mesos agent are equal or greater than what we
+        # did with the `dd` command (i.e., 10 and 10240), because:
+        #   1. Besides the disk writes done by the `dd` command, the statistics may also include
+        #      some disk reads, e.g., to load the necessary executable binary and libraries.
+        #   2. In the environment where RAID is enabled, there may be multiple disk writes to
+        #      different disks for a single `dd` write.
```

Comment:
Did we confirm that this does _not_ pass when we _don't_ do `dd if=/dev/zero of=file bs=1024 count=10 oflag=dsync`?

Reply:
Yeah, I have confirmed it. If I remove `dd if=/dev/zero of=file bs=1024 count=10 oflag=dsync`, the test will fail:
```
>           assert int(total_io_serviced) >= 10, ('Total blkio throttling IO serviced for app_id {} '
                                                  'are less than 10'.format(app_id))
E           AssertionError: Total blkio throttling IO serviced for app_id integration-test-a07ed659cd184ccc87a99635615f6fbc are less than 10
E           assert 1 >= 10
E            +  where 1 = int(1)

test_mesos.py:261: AssertionError
```